tests/engine/processes/workchains/test_restart.py

```diff
@@ -8,44 +8,94 @@
 # For further information please visit http://www.aiida.net               #
 ###########################################################################
 """Tests for `aiida.engine.processes.workchains.restart` module."""
-from aiida.backends.testbase import AiidaTestCase
-from aiida.engine.processes.workchains.restart import BaseRestartWorkChain
-from aiida.engine.processes.workchains.utils import process_handler
+# pylint: disable=invalid-name
+import pytest
 
+from aiida.engine import CalcJob, BaseRestartWorkChain, process_handler, ProcessState, ProcessHandlerReport, ExitCode
```

Comment:
So, this is a more general question.

I understand that from a core-user perspective, having the important methods available through the higher level modules insted of the specific ones (I mean using `from aiida.engine import process_handler` instead of using the complete `from aiida.engine.processes.workchains.utils import process_handler`) is helpful because they don't need to know the internal organization to have access to these.

However, wouldn't still be better for us to internally keep using the most specific path possible when importing inside aiida-core? I mean, I feel that if we need to debug or stuff like that it would be easier to know the exact location of the functions and classes being imported, and although I know most GUI's have some sort of automatic seach engine for this, but perhaps sometimes you need to use some simplified interface or maybe you are just grepping and it helps to have the information there.

Reply:
As you said, most proper IDE's can automatically jump to the source and the increased brevity is a big plus in this case. Exception, in my opinion, is when you are importing something from a submodule. In that case the `from .module import something` is better than the absolute form

Reply:
Additional point, it's important for us to use the "public" interface as much as possible in the tests - if only to make sure that we don't accidentally break it.

Reply:
Ah, ok. Simultaneously checking in the tests that you are not breaking the public interface is a good point, I didn't thought about that. I still think there is some advantages of knowing the exact location of the code you are using, and (at least for the non-test internals) it is not clear to me that the brevity in the "header" of the file (where there is not much logic to obfuscate) outweights this.

Reply:
I really don't think using `from aiida.orm.nodes.data.int import Int` every time instead of `from aiida.orm import Int` does the understanding of the code any favors